### PR TITLE
Added support for Linux Alpine distribution

### PR DESCRIPTION
**Pull request by aleksy-zalenski**

### CODE REVIEW

**Comment by aleksy-zalenski** [comment]
Hello folks,

The following change allows using the cronlock script on Linux Alpine distribution. Unfortunately the timeout command structure is different on Alpine when compered to e.g. Ubuntu, hence the script was not working as expected.

The structure of the timeout command for Alpine Linux is as follows: 

bash-4.4# timeout
BusyBox v1.28.4 (2018-12-06 15:13:21 UTC) multi-call binary.

Usage: timeout [-t SECS] [-s SIG] PROG ARGS

Runs PROG. Sends SIG to it if it is not gone in SECS seconds.
Defaults: SECS: 10, SIG: TERM.

Due to the -t being required for timeout value, we need to change the structure of the command.

Let me know in case of any questions!

**Comment by mitermayer** [comment]
Any chance we can get this merged soon ?

**Comment by kvz** [comment]
Hi, thanks for the contribution, somehow this slipped through the cracks, I'm sorry about that. I think we could merge this, but i have a few nitpicks that i'd like to see taken care of:

- Useless cat, something like `grep -E "^ID=" /etc/os-release | cut -d "=" -f2` or `awk -F= '/^ID=/ {print $2}' /etc/os-release` would be better
- `get_unix_distro` is a function that sets a global, instead it could also return the distro I think. Maybe we don't even need the function at all if it's just an `awk` command
- Would there be a way to change the arguments for `timeout` based on the timeout command, instead of the Alpine distro? I bet there are other platforms out there that pack the same version? Should we check the `-v` of timeout instead, or similar?

**Comment by jbienkowski311** [comment]
@kvz for sure there is some better way to implement this. I have done some digging:

1. Alpine is using BusyBox's timeout command, which has additional `-t` parameter.
2. It is a case for BusyBox versions up to 1.29 (I've checked all Alpine versions until 2.6)
3. It seems that what we could do is run something like `readlink $(which timeout)` and check if it has `busybox`.
4. The only problem is that in the newest Alpine (version 4), BusyBox is in the version 1.30. And the `-t` parameter is gone.

To sum up, more general solution could look like this:
1. Check `readlink $(which timeout)`
2. If it is `/bin/busybox`, then check it's version
3. If the version is less than 1.30, then add `-t` parameter to the `timeoutcmd`

**Comment by kvz** [comment]
Sounds fair, and good!

**Comment by jbienkowski311** [comment]
Please check #15
